### PR TITLE
remove redundant connection param

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by tmasternak** [comment]
@SimonCropp thanks for the PR. 

I'll look into your PR as soon as I can but first I need to make sure our develop build is green again (some deps moved under our feet).

**Comment by tmasternak** [comment]
@SzymonPobiega I say :shipit:.

**Comment by SimonCropp** [comment]
on another note. is that method missing some guards?

File: src/NServiceBus.SqlServer.UnitTests/APIApprovals.Approve.approved.txt [hunk 1]
```diff
@@ -24,7 +24,10 @@ namespace NServiceBus.Transport.SQLServer
     }
     public class static SendOptionsExtensions
     {
+        [System.ObsoleteAttribute("The connection parameter is no longer required. Use the alternate overload with o" +
+            "nly the transaction parameter. Will be removed in version 5.0.0.", true)]
         public static void UseCustomSqlConnectionAndTransaction(this NServiceBus.SendOptions options, System.Data.SqlClient.SqlConnection connection, System.Data.SqlClient.SqlTransaction transaction) { }
+        public static void UseCustomSqlConnectionAndTransaction(this NServiceBus.SendOptions options, System.Data.SqlClient.SqlTransaction transaction) { }
```
Comment (SzymonPobiega) [inline src/NServiceBus.SqlServer.UnitTests/APIApprovals.Approve.approved.txt]: Shouldn't we call it `UseCustomSqlTransaction` now?

**Pull request merged**